st3/lsp_utils/_client_handler/abstract_plugin.py

```diff
@@ -124,6 +124,15 @@ def can_start(cls, window: sublime.Window, initiating_view: sublime.View,
             configuration.command = cls.get_command()
         return None
 
+    @classmethod
+    def on_pre_start(cls, window: sublime.Window, initiating_view: sublime.View,
+                     workspace_folders: List[WorkspaceFolder], configuration: ClientConfig) -> Optional[str]:
+        extra_paths = path.pathsep.join(cls.get_additional_paths())
+        if extra_paths:
+            original_path = configuration.env.get('PATH') or ''
+            configuration.env['PATH'] = path.pathsep.join([extra_paths, original_path])
```

Comment:
Can Node that is locally installed clash with the system installed Node.JS?

For example if a user has local Node installed in ST, and a global Node in the system,
which one will be chosen to run the `node` command?

Reply:
The path is only modified for the language server subprocess.

Reply:
Even in the subprocess, I was curious to find out if two same named commands exist in the path, which one will win... Answer: Apparently the first one that is matched:

https://superuser.com/questions/506450/if-two-separate-path-directories-contain-a-same-named-executable-how-does-windo